deebotozmo/__init__.py

```diff
@@ -346,375 +459,508 @@ def __init__(self, user, resource, secret, vacuum, country, continent, live_map_
             _LOGGER.warning("*** Error type: " + err.etype)
             _LOGGER.warning("*** Error condition: " + err.condition)
 
-    def request_all_statuses(self):
-        if not self.thread_statuses.is_alive():
-            self.thread_statuses = threading.Thread(target=self.refresh_statuses, daemon=False, name="schedule_thread_statuses")
-            self.thread_statuses.start()
+    def refresh_stats(self):
+        try:
+            _LOGGER.debug("[refresh_stats] Begin")
+            self.exc_command("getStats")
+        except XMPPError as err:
+            _LOGGER.warning(
+                "Initial Stats requests failed to reach VacBot. Will try again on next ping."
+            )
+            _LOGGER.warning("*** Error type: " + err.etype)
+            _LOGGER.warning("*** Error condition: " + err.condition)
 
-        if self.live_map_enabled:
-            if not self.thread_livemap.is_alive():
-                self.thread_livemap = threading.Thread(target=self.refresh_liveMap, daemon=False, name="schedule_thread_livemap")
-                self.thread_livemap.start()
-                
-        if not self.thread_components.is_alive():
-            self.thread_components = threading.Thread(target=self.refresh_components, daemon=False, name="schedule_thread_components")
-            self.thread_components.start()
-    
-    def setScheduleUpdates(self, livemap_cycle = 15, status_cycle = 30, components_cycle = 60):
-        # It will refresh all statuses very X seconds
-        if self.live_map_enabled:
-            self.json.schedule(livemap_cycle, self.refresh_liveMap)
+    def refresh_cleanLogs(self):
+        try:
+            _LOGGER.debug("[refresh_cleanLogs] Begin")
+            self.exc_command("GetCleanLogs")
+        except XMPPError as err:
+            _LOGGER.warning(
+                "Initial CleanLogs requests failed to reach VacBot. Will try again on next ping."
+            )
+            _LOGGER.warning("*** Error type: " + err.etype)
+            _LOGGER.warning("*** Error condition: " + err.condition)
+
+    def refresh_liveMap(self, force=False):
+        try:
+            if self.vacuum_status == "STATE_CLEANING" or force == True:
+                _LOGGER.debug("[refresh_liveMap] Begin")
+                self.exc_command("getMapTrace", {"pointCount": 200, "traceStart": 0})
+                self.exc_command("getPos", ["chargePos", "deebotPos"])
+                self.exc_command("getMajorMap")
+                self.live_map = self.__map.GetBase64Map()
+
+                self.livemapEvents.notify(self.live_map)
+        except XMPPError as err:
+            _LOGGER.warning(
+                "Initial live map failed to reach VacBot. Will try again on next ping."
+            )
+            _LOGGER.warning("*** Error type: " + err.etype)
+            _LOGGER.warning("*** Error condition: " + err.condition)
+
+    ###################################################################
+
+    def setScheduleUpdates(
```

Comment:
I would remove this method completely.
As we have already the different events, then we should schedule only them, when some one is subscribed on the event.
Example components:
We should schedule only `self.json.schedule(components_cycle, self.refresh_components)`, when some is subscribed for the liveSpan events, otherwise we are requesting useless information. This will also decrease the number of requests. What do you think?

My idea was something like:
```
class EventEmitterWithSchedule(EventEmitter):
    """A event emitter, which starts a scheduler when at least one subscriber is subscribed"""

    def __init__(self, json: EcoVacsJSON, seconds: int, timer_function):
        super().__init__()
        self._json = json
        self._seconds = seconds
        self._timer_function = timer_function

    def subscribe(self, callback) -> EventListener:
        if len(self._subscribers) == 0:
            self._json.schedule(self._seconds, self._timer_function)
        return super().subscribe(callback)

....
```

Reply:
@And3rsL what do you think?

Reply:
I think it's a cleaner way to do it :) i need some time to code it